[Bug]: Plugin fails to patch build.gradle in all cases

### Mapbox Implementation

Mapbox

### Mapbox Version

default

### Platform

Android

### `@rnmapbox/maps` version

10.0.0-beta.56

### Standalone component to reproduce


```
Could not find method maven() for arguments
```

<img width="1063" alt="image" src="https://user-images.githubusercontent.com/397632/200970635-d7e8701a-27f7-43af-9d6c-c94e2dcb74bd.png">

### Given input like so:

```gradle
// buildscript { ... }
// ...

def REACT_NATIVE_VERSION = new File(['node', '--print',"JSON.parse(require('fs').readFileSync(require.resolve('react-native/package.json'), 'utf-8')).version"].execute(null, rootDir).text.trim())

allprojects {
    configurations.all {
        resolutionStrategy {
            force "com.facebook.react:react-native:" + REACT_NATIVE_VERSION
        }
    }

    repositories {
        mavenLocal()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url(new File(['node', '--print', "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim(), '../android'))
        }
        maven {
            // Android JSC is installed from npm
            url(new File(['node', '--print', "require.resolve('jsc-android/package.json')"].execute(null, rootDir).text.trim(), '../dist'))
        }

        google()
        mavenCentral {
            // We don't want to fetch react-native from Maven Central as there are
            // older versions over there.
            content {
                excludeGroup 'com.facebook.react'
            }
        }
        maven { url 'https://www.jitpack.io' }
    }
}
```

### We get mangled output

```gradle
allprojects {
// @generated begin @rnmapbox/maps-v2-maven - expo prebuild (DO NOT MODIFY) sync-2bc3dfdedd6d41c532c4945588823cd7e6b0a736
repositories {
      maven {
    url 'https://api.mapbox.com/downloads/v2/releases/maven'
    authentication { basic(BasicAuthentication) }
    credentials {
      username = 'mapbox'
      password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: \\"\\"
    }
  }
  }
// @generated end @rnmapbox/maps-v2-maven
    configurations.all {
        resolutionStrategy {
            force \\"com.facebook.react:react-native:\\" + REACT_NATIVE_VERSION
        }
    }

    repositories {
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url(new File(['node', '--print', \\"require.resolve('react-native/package.json')\\"].execute(null, rootDir).text.trim(), '../android'))
        }
        maven {
            // Android JSC is installed from npm
            url(new File(['node', '--print', \\"require.resolve('jsc-android/package.json')\\"].execute(null, rootDir).text.trim(), '../dist'))
        }

        google()
        mavenCentral {
            // We don't want to fetch react-native from Maven Central as there are
            // older versions over there.
            content {
                excludeGroup 'com.facebook.react'
            }
        }
        maven { url 'https://www.jitpack.io' }
    }
}
```

### Observed behavior and steps to reproduce

_No response_

### Expected behavior

`withMapbox` should inspect the input and calculate the line offset to reliably inject content.

### Notes / preliminary analysis

- Adds tests.
- Works in brand new expo app, as well as all my prior projects.


### Additional links and references


```text
.......
w: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/TaskConfiguration.kt: (252, 14): 'BaseVariant' is deprecated. Deprecated in Java
> Task :react-native-gradle-plugin:compileJava
[stderr] Note: /home/expo/workingdir/build/node_modules/react-native-gradle-plugin/src/main/java/com/facebook/react/codegen/generator/SchemaJsonParser.java uses or overrides a deprecated API.
[stderr] Note: Recompile with -Xlint:deprecation for details.
> Task :react-native-gradle-plugin:classes
> Task :react-native-gradle-plugin:inspectClassesForKotlinIC
> Task :react-native-gradle-plugin:jar
> Configure project :app
Unable to detect AGP versions for included builds. All projects in the build should use the same AGP version. Class name for the included build object: org.gradle.composite.internal.DefaultIncludedBuild$IncludedBuildImpl_Decorated.
[stderr] FAILURE: Build failed with an exception.
[stderr] * Where:
[stderr] Build file '/home/expo/workingdir/build/android/build.gradle' line: 43
[stderr] * What went wrong:
[stderr] A problem occurred evaluating project ':app'.
[stderr] > Failed to apply plugin class 'org.gradle.api.plugins.BasePlugin'.
[stderr]    > Could not find method maven() for arguments [build_a4pmw2504rhdm3lg68kv7pbd1$_run_closure1$_closure2$_closure4@5d61668e] on configuration ':app:archives' of type org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] * Get more help at https://help.gradle.org
[stderr] BUILD FAILED in 1m 38s
6 actionable tasks: 6 executed
Error: Gradle build failed with unknown error. See logs for the "Run gradlew" phase for more information.
```

--- a/plugin/build/withMapbox.d.ts
+++ b/plugin/build/withMapbox.d.ts
@@ -13,5 +13,7 @@ export declare function addMapboxInstallerBlock(src: string, blockName: Installe
  * Without this, production builds targeting simulators will fail.
  */
 export declare function setExcludedArchitectures(project: XcodeProject): XcodeProject;
+declare const addMapboxMavenRepo: (projectBuildGradle: string) => string;
 declare const _default: ConfigPlugin<MapboxPlugProps>;
 export default _default;
+export { addMapboxMavenRepo as _addMapboxMavenRepo, };

--- a/plugin/build/withMapbox.js
+++ b/plugin/build/withMapbox.js
@@ -3,7 +3,7 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
     return (mod && mod.__esModule) ? mod : { "default": mod };
 };
 Object.defineProperty(exports, "__esModule", { value: true });
-exports.setExcludedArchitectures = exports.addMapboxInstallerBlock = exports.addInstallerBlock = exports.addConstantBlock = exports.applyCocoaPodsModifications = void 0;
+exports._addMapboxMavenRepo = exports.setExcludedArchitectures = exports.addMapboxInstallerBlock = exports.addInstallerBlock = exports.addConstantBlock = exports.applyCocoaPodsModifications = void 0;
 const fs_1 = require("fs");
 const path_1 = __importDefault(require("path"));
 const config_plugins_1 = require("@expo/config-plugins");
@@ -213,42 +213,72 @@ const addMapboxMavenRepo = (projectBuildGradle) => {
         return projectBuildGradle;
     /*
     Should look like this:
-      allprojects {
-        // ...
-        repositories {
+    
+    allprojects {
+      configurations.all {
+          resolutionStrategy {
+              force \\"com.facebook.react:react-native:\\" + REACT_NATIVE_VERSION
+          }
+      }
+  
+      repositories {
           maven {
             url 'https://api.mapbox.com/downloads/v2/releases/maven'
             authentication { basic(BasicAuthentication) }
             credentials {
               username = 'mapbox'
-              password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: ""
+              password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: \\"\\"
             }
           }
-        // ...
+          mavenLocal()
+          maven {
+              // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+              url(new File(['node', '--print', \\"require.resolve('react-native/package.json')\\"].execute(null, rootDir).text.trim(), '../android'))
+          }
+          // ...
     */
     /*
       Since mergeContents checks the anchor for each line, we can't do a "correct"
-      RegExp for allprojects...repositories and have to insert a second repositories
-      for now
+      RegExp for allprojects...repositories.
+      Instead, we check for the first line of `mavenLocal` or `allprojects`, and then insert the block after some content inspection.
     */
+    let offset = 1;
+    let anchor;
+    const allProjectsPattern = new RegExp(`^\\s*allprojects\\s*{`, 'gm');
+    const mavenLocalPattern = new RegExp(`^\\s*mavenLocal\\(\\)`, 'gm');
+    const mavenLocalMatch = projectBuildGradle.match(mavenLocalPattern);
+    if (mavenLocalMatch) {
+        mavenLocalPattern.lastIndex = 0;
+        anchor = mavenLocalPattern;
+        offset = 1;
+    }
+    else {
+        // hack to count offset
+        const allProjectSplit = projectBuildGradle.split(allProjectsPattern);
+        const allProjectLines = allProjectSplit[allProjectSplit.length - 1].split('\n');
+        const allProjectReposOffset = allProjectLines.findIndex((line) => line.includes('repositories'));
+        allProjectsPattern.lastIndex = 0;
+        anchor = allProjectsPattern;
+        offset = allProjectReposOffset + 1;
+    }
     return (0, generateCode_1.mergeContents)({
         tag: `@rnmapbox/maps-v2-maven`,
         src: projectBuildGradle,
-        newSrc: `repositories {
-    maven {
-  url 'https://api.mapbox.com/downloads/v2/releases/maven'
-  authentication { basic(BasicAuthentication) }
-  credentials {
-    username = 'mapbox'
-    password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: ""
-  }
-}
-}`,
-        anchor: new RegExp(`^\\s*allprojects\\s*{`),
-        offset: 1,
+        newSrc: `
+        maven {
+          url 'https://api.mapbox.com/downloads/v2/releases/maven'
+          authentication { basic(BasicAuthentication) }
+          credentials {
+            username = 'mapbox'
+            password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: ""
+          }
+        }\n`,
+        anchor,
+        offset,
         comment: '//',
     }).contents;
 };
+exports._addMapboxMavenRepo = addMapboxMavenRepo;
 const withAndroidAppGradle = (config) => {
     return (0, config_plugins_1.withAppBuildGradle)(config, ({ modResults, ...config }) => {
         if (modResults.language !== 'groovy') {

--- a/plugin/build/withMapbox.js
+++ b/plugin/build/withMapbox.js
@@ -227,7 +227,7 @@ const addMapboxMavenRepo = (projectBuildGradle) => {
             authentication { basic(BasicAuthentication) }
             credentials {
               username = 'mapbox'
-              password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: \\"\\"
+              password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: ""
             }
           }
           mavenLocal()

--- a/plugin/build/withMapbox.js
+++ b/plugin/build/withMapbox.js
@@ -240,27 +240,18 @@ const addMapboxMavenRepo = (projectBuildGradle) => {
     /*
       Since mergeContents checks the anchor for each line, we can't do a "correct"
       RegExp for allprojects...repositories.
-      Instead, we check for the first line of `mavenLocal` or `allprojects`, and then insert the block after some content inspection.
+      Instead, we check for the first `allprojects`, and then count the # of lines to the next `repositories` block.
     */
-    let offset = 1;
-    let anchor;
-    const allProjectsPattern = new RegExp(`^\\s*allprojects\\s*{`, 'gm');
-    const mavenLocalPattern = new RegExp(`^\\s*mavenLocal\\(\\)`, 'gm');
-    const mavenLocalMatch = projectBuildGradle.match(mavenLocalPattern);
-    if (mavenLocalMatch) {
-        mavenLocalPattern.lastIndex = 0;
-        anchor = mavenLocalPattern;
-        offset = 1;
-    }
-    else {
-        // hack to count offset
-        const allProjectSplit = projectBuildGradle.split(allProjectsPattern);
-        const allProjectLines = allProjectSplit[allProjectSplit.length - 1].split('\n');
-        const allProjectReposOffset = allProjectLines.findIndex((line) => line.includes('repositories'));
-        allProjectsPattern.lastIndex = 0;
-        anchor = allProjectsPattern;
-        offset = allProjectReposOffset + 1;
-    }
+    let offset = 0;
+    const anchor = new RegExp(`^\\s*allprojects\\s*{`, 'gm');
+    // hack to count offset
+    const allProjectSplit = projectBuildGradle.split(anchor);
+    if (allProjectSplit.length <= 1)
+        throw new Error('Could not find `allprojects` block');
+    const allProjectLines = allProjectSplit[allProjectSplit.length - 1].split('\n');
+    const allProjectReposOffset = allProjectLines.findIndex((line) => line.includes('repositories'));
+    anchor.lastIndex = 0;
+    offset = allProjectReposOffset + 1;
     return (0, generateCode_1.mergeContents)({
         tag: `@rnmapbox/maps-v2-maven`,
         src: projectBuildGradle,

--- a/plugin/src/__tests__/fixtures/buildGradleFiles.ts
+++ b/plugin/src/__tests__/fixtures/buildGradleFiles.ts
@@ -1,0 +1,73 @@
+export const expoTemplateBuildGradle = `
+import org.apache.tools.ant.taskdefs.condition.Os
+
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {
+    ext {
+        buildToolsVersion = findProperty('android.buildToolsVersion') ?: '31.0.0'
+        minSdkVersion = Integer.parseInt(findProperty('android.minSdkVersion') ?: '21')
+        compileSdkVersion = Integer.parseInt(findProperty('android.compileSdkVersion') ?: '31')
+        targetSdkVersion = Integer.parseInt(findProperty('android.targetSdkVersion') ?: '31')
+        if (findProperty('android.kotlinVersion')) {
+            kotlinVersion = findProperty('android.kotlinVersion')
+        }
+        frescoVersion = findProperty('expo.frescoVersion') ?: '2.5.0'
+
+        if (System.properties['os.arch'] == 'aarch64') {
+            // For M1 Users we need to use the NDK 24 which added support for aarch64
+            ndkVersion = '24.0.8215888'
+        } else {
+            // Otherwise we default to the side-by-side NDK version from AGP.
+            ndkVersion = '21.4.7075529'
+        }
+    }
+    repositories {
+        google()
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'com.google.gms:google-services:4.3.3'
+        classpath('com.android.tools.build:gradle:7.0.4')
+        classpath('com.facebook.react:react-native-gradle-plugin')
+        classpath('de.undercouch:gradle-download-task:4.1.2')
+        // NOTE: Do not place your application dependencies here; they belong
+        // in the individual module build.gradle files
+    }
+}
+
+def REACT_NATIVE_VERSION = new File(['node', '--print',"JSON.parse(require('fs').readFileSync(require.resolve('react-native/package.json'), 'utf-8')).version"].execute(null, rootDir).text.trim())
+
+allprojects {
+    configurations.all {
+        resolutionStrategy {
+            force "com.facebook.react:react-native:" + REACT_NATIVE_VERSION
+        }
+    }
+
+    repositories {
+        mavenLocal()
+        maven {
+            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+            url(new File(['node', '--print', "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim(), '../android'))
+        }
+        maven {
+            // Android JSC is installed from npm
+            url(new File(['node', '--print', "require.resolve('jsc-android/package.json')"].execute(null, rootDir).text.trim(), '../dist'))
+        }
+
+        google()
+        mavenCentral {
+            // We don't want to fetch react-native from Maven Central as there are
+            // older versions over there.
+            content {
+                excludeGroup 'com.facebook.react'
+            }
+        }
+        maven { url 'https://www.jitpack.io' }
+    }
+}
+`;
+
+export const expoTemplateBuildGradleWithoutMavenLocal =
+  expoTemplateBuildGradle.replace(`        mavenLocal()\n`, ``);

--- a/plugin/src/__tests__/withMapbox-test.ts
+++ b/plugin/src/__tests__/withMapbox-test.ts
@@ -1,26 +1,44 @@
-import { applyCocoaPodsModifications } from '../withMapbox';
+import {
+  applyCocoaPodsModifications,
+  _addLibCppFilter,
+  _addMapboxMavenRepo,
+} from '../withMapbox';
 
-import * as fixtures from './fixtures/cocoapodFiles';
+import * as iosFixtures from './fixtures/cocoapodFiles';
+import * as androidFixtures from './fixtures/buildGradleFiles';
+
+describe('applyAndroidGradleModifications', () => {
+  it(`adds the correct repo to build.gradle`, () => {
+    const result = _addMapboxMavenRepo(androidFixtures.expoTemplateBuildGradle);
+    expect(result).toMatchSnapshot();
+  });
+  it(`adds the correct maven repo under allProjects to build.gradle`, () => {
+    const result = _addMapboxMavenRepo(
+      androidFixtures.expoTemplateBuildGradleWithoutMavenLocal,
+    );
+    expect(result).toMatchSnapshot();
+  });
+});
 
 describe('applyCocoaPodsModifications', () => {
   it('adds blocks to a react native template podfile', () => {
     expect(
-      applyCocoaPodsModifications(fixtures.reactNativeTemplatePodfile, {}),
+      applyCocoaPodsModifications(iosFixtures.reactNativeTemplatePodfile, {}),
     ).toMatchSnapshot();
   });
   it('adds blocks to a expo prebuild template podfile', () => {
     expect(
-      applyCocoaPodsModifications(fixtures.expoTemplatePodfile, {}),
+      applyCocoaPodsModifications(iosFixtures.expoTemplatePodfile, {}),
     ).toMatchSnapshot();
   });
   it('adds blocks to a expo prebuild template podfile with custom modifications', () => {
     expect(
-      applyCocoaPodsModifications(fixtures.customExpoTemplatePodfile, {}),
+      applyCocoaPodsModifications(iosFixtures.customExpoTemplatePodfile, {}),
     ).toMatchSnapshot();
   });
   it('fails to add blocks to a bare podfile', () => {
     expect(() =>
-      applyCocoaPodsModifications(fixtures.blankTemplatePodfile, {}),
+      applyCocoaPodsModifications(iosFixtures.blankTemplatePodfile, {}),
     ).toThrow('Failed to match');
     expect(() => applyCocoaPodsModifications('', {})).toThrow(
       'Failed to match',
@@ -28,7 +46,7 @@ describe('applyCocoaPodsModifications', () => {
   });
   it('does not re add blocks to an applied template podfile', () => {
     const runOnce = applyCocoaPodsModifications(
-      fixtures.reactNativeTemplatePodfile,
+      iosFixtures.reactNativeTemplatePodfile,
       {},
     );
 
@@ -36,7 +54,7 @@ describe('applyCocoaPodsModifications', () => {
   });
   it('works after revisions to blocks', () => {
     const runOnce = applyCocoaPodsModifications(
-      fixtures.expoTemplateWithRevisions,
+      iosFixtures.expoTemplateWithRevisions,
       {},
     );
 
@@ -46,7 +64,7 @@ describe('applyCocoaPodsModifications', () => {
   // has a pre_install/post_install blocmk commented out, before the `use_react_native` function.
   it('does not work with revisions to blocks after comments', () => {
     const runOnce = applyCocoaPodsModifications(
-      fixtures.expoTemplateWithRevisionsAfterComments,
+      iosFixtures.expoTemplateWithRevisionsAfterComments,
       {},
     );
 

--- a/plugin/src/__tests__/withMapbox-test.ts
+++ b/plugin/src/__tests__/withMapbox-test.ts
@@ -1,6 +1,5 @@
 import {
   applyCocoaPodsModifications,
-  _addLibCppFilter,
   _addMapboxMavenRepo,
 } from '../withMapbox';
 
@@ -12,11 +11,25 @@ describe('applyAndroidGradleModifications', () => {
     const result = _addMapboxMavenRepo(androidFixtures.expoTemplateBuildGradle);
     expect(result).toMatchSnapshot();
   });
+
   it(`adds the correct maven repo under allProjects to build.gradle`, () => {
     const result = _addMapboxMavenRepo(
       androidFixtures.expoTemplateBuildGradleWithoutMavenLocal,
     );
     expect(result).toMatchSnapshot();
+  });
+
+  it('throws on missing allprojects', () => {
+    expect(() => {
+      _addMapboxMavenRepo(`
+      repositories {
+        mavenLocal()
+        maven {
+            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+            url(new File(['node', '--print', "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim(), '../android'))
+        }
+      }`);
+    }).toThrow();
   });
 });
 

--- a/plugin/src/withMapbox.ts
+++ b/plugin/src/withMapbox.ts
@@ -274,41 +274,72 @@ const addMapboxMavenRepo = (projectBuildGradle: string): string => {
 
   /*
   Should look like this:
-    allprojects {
-      // ...
-      repositories {
+  
+  allprojects {
+    configurations.all {
+        resolutionStrategy {
+            force \\"com.facebook.react:react-native:\\" + REACT_NATIVE_VERSION
+        }
+    }
+
+    repositories {
         maven {
           url 'https://api.mapbox.com/downloads/v2/releases/maven'
           authentication { basic(BasicAuthentication) }
-          credentials { 
+          credentials {
             username = 'mapbox'
-            password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: ""
+            password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: \\"\\"
           }
         }
-      // ...
+        mavenLocal()
+        maven {
+            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+            url(new File(['node', '--print', \\"require.resolve('react-native/package.json')\\"].execute(null, rootDir).text.trim(), '../android'))
+        }
+        // ...
   */
 
   /*
     Since mergeContents checks the anchor for each line, we can't do a "correct"
-    RegExp for allprojects...repositories and have to insert a second repositories
-    for now
+    RegExp for allprojects...repositories.
+    Instead, we check for the first line of `mavenLocal` or `allprojects`, and then insert the block after some content inspection.
   */
+  let offset = 1;
+  let anchor: RegExp;
+  const allProjectsPattern = new RegExp(`^\\s*allprojects\\s*{`, 'gm');
+  const mavenLocalPattern = new RegExp(`^\\s*mavenLocal\\(\\)`, 'gm');
+  const mavenLocalMatch = projectBuildGradle.match(mavenLocalPattern);
+  if (mavenLocalMatch) {
+    mavenLocalPattern.lastIndex = 0;
+    anchor = mavenLocalPattern;
+    offset = 1;
+  } else {
+    // hack to count offset
+    const allProjectSplit = projectBuildGradle.split(allProjectsPattern);
+    const allProjectLines =
+      allProjectSplit[allProjectSplit.length - 1].split('\n');
+    const allProjectReposOffset = allProjectLines.findIndex((line) =>
+      line.includes('repositories'),
+    );
+    allProjectsPattern.lastIndex = 0;
+    anchor = allProjectsPattern;
+    offset = allProjectReposOffset + 1;
+  }
 
   return mergeContents({
     tag: `@rnmapbox/maps-v2-maven`,
     src: projectBuildGradle,
-    newSrc: `repositories {
-    maven {
-  url 'https://api.mapbox.com/downloads/v2/releases/maven'
-  authentication { basic(BasicAuthentication) }
-  credentials {
-    username = 'mapbox'
-    password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: ""
-  }
-}
-}`,
-    anchor: new RegExp(`^\\s*allprojects\\s*{`),
-    offset: 1,
+    newSrc: `
+        maven {
+          url 'https://api.mapbox.com/downloads/v2/releases/maven'
+          authentication { basic(BasicAuthentication) }
+          credentials {
+            username = 'mapbox'
+            password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: ""
+          }
+        }\n`,
+    anchor,
+    offset,
     comment: '//',
   }).contents;
 };
@@ -372,3 +403,17 @@ const withMapbox: ConfigPlugin<MapboxPlugProps> = (
 };
 
 export default createRunOncePlugin(withMapbox, pkg.name, pkg.version);
+
+// TODO: export internal functions for testing purposes
+export {
+  // the following methods accept a string and return a string
+  addMapboxMavenRepo as _addMapboxMavenRepo,
+  // addLibCppFilter as _addLibCppFilter,
+  // Following methods accept a config object
+  // withAndroidProperties as _withAndroidProperties,
+  // withAndroidPropertiesDownloadToken as _withAndroidPropertiesDownloadToken,
+  // withAndroidPropertiesImpl2 as _withAndroidPropertiesImpl2,
+  // withAndroidAppGradle as _withAndroidAppGradle,
+  // withAndroidProjectGradle as _withAndroidProjectGradle,
+  // withMapboxAndroid as _withMapboxAndroid,
+};

--- a/plugin/src/withMapbox.ts
+++ b/plugin/src/withMapbox.ts
@@ -302,29 +302,22 @@ const addMapboxMavenRepo = (projectBuildGradle: string): string => {
   /*
     Since mergeContents checks the anchor for each line, we can't do a "correct"
     RegExp for allprojects...repositories.
-    Instead, we check for the first line of `mavenLocal` or `allprojects`, and then insert the block after some content inspection.
+    Instead, we check for the first `allprojects`, and then count the # of lines to the next `repositories` block.
   */
-  let offset = 1;
-  let anchor: RegExp;
-  const allProjectsPattern = new RegExp(`^\\s*allprojects\\s*{`, 'gm');
-  const mavenLocalPattern = new RegExp(`^\\s*mavenLocal\\(\\)`, 'gm');
-  const mavenLocalMatch = projectBuildGradle.match(mavenLocalPattern);
-  if (mavenLocalMatch) {
-    mavenLocalPattern.lastIndex = 0;
-    anchor = mavenLocalPattern;
-    offset = 1;
-  } else {
-    // hack to count offset
-    const allProjectSplit = projectBuildGradle.split(allProjectsPattern);
-    const allProjectLines =
-      allProjectSplit[allProjectSplit.length - 1].split('\n');
-    const allProjectReposOffset = allProjectLines.findIndex((line) =>
-      line.includes('repositories'),
-    );
-    allProjectsPattern.lastIndex = 0;
-    anchor = allProjectsPattern;
-    offset = allProjectReposOffset + 1;
-  }
+  let offset = 0;
+  const anchor = new RegExp(`^\\s*allprojects\\s*{`, 'gm');
+  // hack to count offset
+  const allProjectSplit = projectBuildGradle.split(anchor);
+  if (allProjectSplit.length <= 1)
+    throw new Error('Could not find `allprojects` block');
+
+  const allProjectLines =
+    allProjectSplit[allProjectSplit.length - 1].split('\n');
+  const allProjectReposOffset = allProjectLines.findIndex((line) =>
+    line.includes('repositories'),
+  );
+  anchor.lastIndex = 0;
+  offset = allProjectReposOffset + 1;
 
   return mergeContents({
     tag: `@rnmapbox/maps-v2-maven`,

--- a/plugin/src/withMapbox.ts
+++ b/plugin/src/withMapbox.ts
@@ -288,7 +288,7 @@ const addMapboxMavenRepo = (projectBuildGradle: string): string => {
           authentication { basic(BasicAuthentication) }
           credentials {
             username = 'mapbox'
-            password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: \\"\\"
+            password = project.properties['MAPBOX_DOWNLOADS_TOKEN'] ?: ""
           }
         }
         mavenLocal()

